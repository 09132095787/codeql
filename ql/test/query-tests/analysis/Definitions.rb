module A
  class B
    def f
      g 1
    end

    def g x
      x
    end

    def h
      f
    end
  end
end

module C
  class D
    def initialize
      @e = 1
      x, y = [1, 2]
      y
    end

    def h y
      A::B.new.g y
      UnknownClass.some_method
      @f = 2
      @e
      @f
    end
  end
end

C::D.new
