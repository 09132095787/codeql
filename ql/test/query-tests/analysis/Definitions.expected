| Definitions.rb:4:7:4:9 | call to g | Definitions.rb:7:5:9:7 | g | method |
| Definitions.rb:8:7:8:7 | x | Definitions.rb:7:11:7:11 | x | variable |
| Definitions.rb:12:7:12:7 | call to f | Definitions.rb:3:5:5:7 | f | method |
| Definitions.rb:20:7:20:7 | A | Definitions.rb:1:1:15:3 | A | constant |
| Definitions.rb:20:7:20:10 | B | Definitions.rb:2:3:14:5 | B | constant |
| Definitions.rb:20:18:20:18 | y | Definitions.rb:19:11:19:11 | y | variable |
| Definitions.rb:26:1:26:1 | C | Definitions.rb:17:1:24:3 | C | constant |
| Definitions.rb:26:1:26:4 | D | Definitions.rb:18:3:23:5 | D | constant |
